primer/src/Primer/API.hs

```diff
@@ -633,6 +636,25 @@ data Module = Module
   deriving (ToJSON, FromJSON) via PrimerJSON Module
   deriving anyclass (NFData)
 
+data TypeDef = TypeDef
+  { name :: TyConName
+  , params :: [TyVarName]
+  , nameHints :: [Name.Name]
+  , constructors :: Maybe [ValCon]
+  -- ^ a `Nothing` here indicates a primitive type (whereas `Just []` is `Void`)
+  }
```

Comment:
Why do these need name hints? Just so a frontend can display them if wanted?

Reply:
Yeah, let's leave these in place. It would be great if students could add their own hints.